rclcpp_examples/src/topics/listener_single_msg.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/executors.hpp>
+#include <rclcpp/memory_strategies.hpp>
+#include <rclcpp/strategies/single_msg_memory_strategy.hpp>
+
+#include <std_interfaces/msg/int32.hpp>
+#include <std_interfaces/msg/string.hpp>
+
+using namespace rclcpp::single_msg_memory_strategy;
+using rclcpp::memory_strategies::static_memory_strategy::StaticMemoryStrategy;
+
+unsigned int messages_received = 0;
+
+void chatterCallback(const std_interfaces::msg::String::ConstSharedPtr & /*msg*/)
+{
+  ++messages_received;
+}
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::memory_strategy::MemoryStrategy::SharedPtr memory_strategy = nullptr;
+  if (argc > 1) {
+    std::string argument(argv[1]);
+    if (argument == "static") {
+      printf("Setting memory allocation strategy to 'static'.\n");
```

Comment:
I'm using `printf` because @dirk-thomas always comments that it should be used instead of `std::cout` to avoid interleaving. I'm not concerned about allocations here because it's not during the execution loop of the node.